src/ert/_c_wrappers/enkf/enkf_state.py

```diff
@@ -43,9 +42,16 @@ def _internalize_GEN_DATA(
     return (LoadStatus.LOAD_SUCCESSFUL, "")
 
 
+def _should_load_summary_key(data_key, user_set_keys):
+    for key in user_set_keys:
+        if fnmatch.fnmatch(data_key, key):
+            return True
+    return False
```

Comment:
Implement the github annotation

```suggestion
    return any(fnmatch.fnmatch(data_key, key) for key in user_set_keys)
```